providers/v1/aws/secretsmanager/secretsmanager.go

```diff
@@ -888,20 +887,9 @@ func (sm *SecretsManager) manageResourcePolicy(ctx context.Context, metadata *ap
 		currentPolicy = *currentPolicyOutput.ResourcePolicy
 	}
 
-	// convert to maps so we can do a stable comparison.
-	var (
-		currentPolicyMap map[string]any
-		policyJSONMaps   map[string]any
-	)
-
-	if err := json.Unmarshal([]byte(currentPolicy), &currentPolicyMap); err != nil {
-		return fmt.Errorf("failed to unmarshal current resource policy: %w", err)
-	}
-	if err := json.Unmarshal([]byte(policyJSON), &policyJSONMaps); err != nil {
-		return fmt.Errorf("failed to unmarshal current resource policy: %w", err)
-	}
-
-	if maps.Equal(currentPolicyMap, policyJSONMaps) {
+	if isSame, err := esutils.CanonicalJSONCompare(currentPolicy, policyJSON); err != nil {
```

Comment:
Rather than using a canonical JSON compare which is an incomplete comparison, how about just type switching? In case of a slice, use a slice, in case of a map use a map?

Reply:
I think this works. Especially if other parts of the code are already using this. And your test pretty much shows that it works, right?